configuration/debian/tedge/preinst

```diff
@@ -1,69 +1,89 @@
 #!/bin/sh
 set -e
 
+command_exists() {
+    command -V "$1" >/dev/null 2>&1
+}
+
 ### Create groups
 if ! getent group tedge >/dev/null; then
-    groupadd --system tedge
+    if command_exists groupadd; then
+        groupadd --system tedge
+    elif command_exists addgroup; then
+        addgroup -S tedge
+    else
+        echo "WARNING: Could not create group: tedge" >&2
```

Comment:
It's just a question. I'm wondering if failing user/group creation of `tedge` is "warning". Indeed people can create it later and if they want to run everything with `root` then no pronlem (e.g. in container). But other components like tedge-mapper will run by the `tedge` user as long as systemd is used. So, in some scenario, no existence of `tedge` is more like "error", although not all scenarios.

Reply:
It is warning here because `tedge init` will most likely fail later on because it tries to create files/folders with the configured owner. But I didn't want the user creation to be coupled with the `tedge init` code, so that in case if we change the handling of `tedge init` to be more relaxed if the user is not present, then we don't have to change the maintainer script.